bluemira/base/reactor_config.py

```diff
@@ -194,15 +194,17 @@ def config_for(self, component_name: str, *args: str) -> dict:
         return _return
 
     @staticmethod
-    def _read_or_return(dict_or_str_path: Union[str, dict]) -> Dict:
-        if isinstance(dict_or_str_path, str):
-            return ReactorConfig._read_json_file(dict_or_str_path)
-        elif isinstance(dict_or_str_path, dict):
-            return dict_or_str_path
-        raise ReactorConfigError("Invalid config_path")
+    def _read_or_return(config_path: Union[str, Path, dict]) -> Dict:
+        if isinstance(config_path, str) or isinstance(config_path, Path):
+            return ReactorConfig._read_json_file(config_path)
+        elif isinstance(config_path, dict):
+            return config_path
+        raise ReactorConfigError(
+            "config_path must be either a dict, a Path object, or a string"
```

Comment:
Sorry, another nitpicky one. I always think it makes ID-ing the problem easier if you print out the bad type.

```suggestion
            f"config_path must be either a dict, a Path object, or a string, found {type(config_path)}."
```

Reply:
Meant to do that, nice catch